File: Difficulty: Medium/Children Sum in a Binary Tree/children-sum-in-a-binary-tree.cpp
/*Complete the function below

struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){80
        data = x;
        left = right = NULL;
    }
};
*/

class Solution {
  public:
    // Function to check whether all nodes of a tree have the value
    // equal to the sum of their child nodes.
    int isSumProperty(Node *root) {
        // Add your code here
        if (root == NULL || (root->left == NULL && root->right == NULL))
            return 1;
        
        int left =0;
        int right = 0;
        if(root->left!=NULL) left = root->left->data;
        if(root->right!=NULL) right = root->right->data;
        
        if(root->data == left+right && isSumProperty(root->left) && isSumProperty(root->right)){
            return 1;
        }
        else{
            return 0;
        }
          
        
    }
};

File: Difficulty: Medium/Children Sum in a Binary Tree/children-sum-in-a-binary-tree.py
'''
# Node Class:
class Node:
    def init(self,val):
        self.data = val
        self.left = None
        self.right = None
'''

class Solution:
    def isSumProperty(self, root):
        # code here
        if root is None or (root.left is None and root.right is None) :
            return True
        
        left=0
        right=0
        if root.left:
            left=root.left.data
        if root.right:
            right=root.right.data
        
        tot=left+right
        if tot==root.data and self.isSumProperty(root.left) and self.isSumProperty(root.right):
            return True
        return False
            
            
    